### ISSUE TITLE
Pytorch model.train() should be called every step instead of only once

### ISSUE THREAD

**Issue by chiragjn** [ThilinaRajapakse/simpletransformers#739]
**Describe the bug**
`model.train()` is being called once before iteration over data starts
https://github.com/ThilinaRajapakse/simpletransformers/blob/ae28b73ebd6bd6a7dc6bfa454c6d9c42185a8494/simpletransformers/classification/classification_model.py#L506-L518
However, if `evaluate_during_training` is enabled, `model.eval()` will be called (via `eval_model` -> `evaluate`) which is correct but the model will not be put back into train mode

**To Reproduce**
N/A

**Expected behavior**
model.train() should be called at every step before model(**inputs)

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS

**Additional context**
N/A

**Comment by ThilinaRajapakse** [comment]
I just made a quick fix for this. But, I'm wondering how this didn't seem to cause any issues.

**Comment by chiragjn** [comment]
Thanks for the fix, to be honest, I didn't notice any problems in my experiments either. But when I started comparing code from `Trainer` interface from `transformers` lib this stood out.
I guess this is the curse of deep learning :( Things work empirically even when they are not supposed to in theory and we have no idea why.
